daemon/network.go

```diff
@@ -315,9 +315,21 @@ func (daemon *Daemon) createNetwork(create types.NetworkCreateRequest, id string
 		driver = c.Config().DefaultDriver
 	}
 
+	networkOptions := create.Options
```

Comment:
Maps are reference types so `networkOptions` is just an alias. The `create.Options` map is still mutated in-place. Either ditch the misleading alias or deep-copy the map.